src/private_position/move_generation.ts

```diff
@@ -143,6 +143,31 @@ export function isStalemate(position: PositionImpl) {
 }
 
 
+/**
+ * Whether the given position is level and there is insufficient material on board.
+ */
+export function isInsufficientMaterial(position: PositionImpl) {
```

Comment:
Unit tests should be suitable for this method, in order to ensure it's behavior is correct for standard and corner cases. For example, one corner case: `k7/n1K5/B7/8/8/8/8/8 w - - 0 1` -> I guess your method returns `true`, while it should return `false` (since there is `Bb7#`).

Also: game variants `NO_KING`, `*_KING_ONLY` and `HORDE` should be handled specifically I think (e.g. at Horde, white cannot checkmate with only a queen).

Reply:
> For example, one corner case: k7/n1K5/B7/8/8/8/8/8 w - - 0 1 -> I guess your method returns true, while it should return false (since there is Bb7#).

Its a bit trickier since black cannot be forced into such position.  

FIDE declares automatic draw due to insufficient materials when no forced mate is possible.
While USCF allows for any mate, even if its by mistake

For example two knights vs lone king is considered draw under USCF as there can be no forced mate, but not under FIDE as mate is still possible by accident.

I've added a flag `isinsufficientMaterial(forcedMate?: boolean)` to allow both options.  
Notes are in the code.  

> Also: game variants NO_KING, *_KING_ONLY and HORDE should be handled specifically I think (e.g. at Horde, white cannot checkmate with only a queen).

My notes explicitly state that method returns always false for **horde** and **antichess**  
This is similar exception you have in some other methods as well

Reply:
I was not aware of the USCF specificity. Having a flag to support both rules is relevant then.